onnxruntime/core/providers/cpu/reduction/reduction_ops.h

```diff
@@ -74,6 +135,9 @@ class ReduceKernelBase {
   bool keepdims_;
   bool noop_with_empty_axes_;
   bool select_last_index_;
+
+  // Caches the configuration of the last execution.
+  mutable ResultsExperimentalPrepareForReduce last_results_;
```

Comment:
I am not sure this will consist with all serving scenarios, maybe caching is not worth for the risk.
    * Framework currently require const, if cache is needed, better work out a api to store it some where else.
    * const tensor checking may handle some case already.

Reply:
I removed it. I did not find any method to cache information in ctx, or maybe in the constructor if the shape of the input is known. Maybe one operator is already doing that. For some deep learning model, I noticed that one Reduce operator could be used multiple time in the graph with the same shape and axis.